spec/lib/cli_spec.rb

```diff
@@ -0,0 +1,36 @@
+module TFA
+  describe CLI do
+    subject { CLI.new }
+
+    def code_for(secret)
+      ::ROTP::TOTP.new(secret).now
+    end
+
+    let(:secret) { ::ROTP::Base32.random_base32 }
+
+    describe "#add" do
+      context "when a secret is added" do
+        it "adds the secret" do
+          subject.add('development', secret)
+          expect(subject.show('development')).to eql(secret)
+        end
+      end
+
+      context "when a full otpauth string is added" do
+        it "strips out the url for just the secret" do
+          subject.add('development', "otpauth://totp/email@email.com?secret=#{secret}&issuer=")
+          expect(subject.show('development')).to eql(secret)
+        end
+      end
+    end
+
+    describe "#totp" do
+      context "when a single key is given" do
+        it "returns a time based one time password for the authentication secret given" do
+          subject.add('development', secret)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.